### PR TITLE
Fixes to the Oculus Quest asmdefs for CI

### PR DESCRIPTION
**Pull request by RogPodge**
## Overview

Removes Oculus.VR references from the asmdefs so that the CI build would work. When the Oculus Integration package is used, the user can use 

Mixed Reality Toolkit > Utilities > Oculus > Integrate Oculus Integration Unity Modules

in editor to add the references back in.

### CODE REVIEW

File: Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Editor/MRTKQuestConfigurationChecker.cs [hunk 1]
```diff
@@ -53,16 +53,65 @@ static class OculusConfigurationChecker
...
+                    if (references.Contains("Oculus.VR.Editor"))
+                    {
+                        references.Remove("Oculus.VR.Editor");
+                    }
+                    oculusAsmDef.References = references.ToArray();
+                }
+                else
+                {
+                    if(oculusAsmDefFile == oculusXRSDKAsmDefFile || oculusAsmDefFile == oculusXRSDKHandtrackingUtilsAsmDefFile)
+                        oculusAsmDef.AddReference("Oculus.VR");
```
Comment (davidkline-ms) [inline Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Editor/MRTKQuestConfigurationChecker.cs:98]: nit: please add { } around the single line ifs

**Comment by CDiaz-MS** [comment]
Getting errors when I open the branch in a 2019.4.7 project:

```
ArgumentException: The Object you want to instantiate is null.
UnityEngine.Object.CheckNullArgument (System.Object arg, System.String message) (at <733c7a7c972a4ba7b7c5fd5886daf788>:0)
UnityEngine.Object.Instantiate[T] (T original) (at <733c7a7c972a4ba7b7c5fd5886daf788>:0)
Microsoft.MixedReality.Toolkit.XRSDK.Oculus.MRTKOculusConfig.get_Instance () (at Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Config/MRTKOculusConfig.cs:46)
Microsoft.MixedReality.Toolkit.XRSDK.Oculus.OculusHand..ctor (Microsoft.MixedReality.Toolkit.TrackingState trackingState, Microsoft.MixedReality.Toolkit.Utilities.Handedness controllerHandedness, Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputSource inputSource, Microsoft.MixedReality.Toolkit.Input.MixedRealityInteractionMapping[] interactions) (at Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Input/Controllers/OculusHand.cs:85)
System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) (at <fb001e01371b4adca20013e0ac763896>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Reflection.MonoCMethod.DoInvoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Reflection.MonoCMethod.Invoke (System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <fb001e01371b4adca20013e0ac763896>:0)
System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Activator.CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Activator.CreateInstance (System.Type type, System.Object[] args) (at <fb001e01371b4adca20013e0ac763896>:0)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMapping.GetDefaultInteractionMappings () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMapping.cs:220)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMapping.SetDefaultInteractionMapping (System.Boolean overwrite) (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMapping.cs:132)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMappingProfile.AddMappings () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMappingProfile.cs:156)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMappingProfile.Awake () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMappingProfile.cs:133)

```

```
ArgumentException: The Object you want to instantiate is null.
UnityEngine.Object.CheckNullArgument (System.Object arg, System.String message) (at <733c7a7c972a4ba7b7c5fd5886daf788>:0)
UnityEngine.Object.Instantiate[T] (T original) (at <733c7a7c972a4ba7b7c5fd5886daf788>:0)
Microsoft.MixedReality.Toolkit.XRSDK.Oculus.MRTKOculusConfig.get_Instance () (at Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Config/MRTKOculusConfig.cs:46)
Microsoft.MixedReality.Toolkit.XRSDK.Oculus.OculusHand..ctor (Microsoft.MixedReality.Toolkit.TrackingState trackingState, Microsoft.MixedReality.Toolkit.Utilities.Handedness controllerHandedness, Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputSource inputSource, Microsoft.MixedReality.Toolkit.Input.MixedRealityInteractionMapping[] interactions) (at Assets/MRTK/Providers/Oculus/XRSDK/MRTK-Quest/Scripts/Input/Controllers/OculusHand.cs:85)
System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) (at <fb001e01371b4adca20013e0ac763896>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Reflection.MonoCMethod.DoInvoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Reflection.MonoCMethod.Invoke (System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <fb001e01371b4adca20013e0ac763896>:0)
System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Activator.CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) (at <fb001e01371b4adca20013e0ac763896>:0)
System.Activator.CreateInstance (System.Type type, System.Object[] args) (at <fb001e01371b4adca20013e0ac763896>:0)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMapping.GetDefaultInteractionMappings () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMapping.cs:220)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMapping.SetDefaultInteractionMapping (System.Boolean overwrite) (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMapping.cs:132)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMappingProfile.AddMappings () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMappingProfile.cs:156)
Microsoft.MixedReality.Toolkit.Input.MixedRealityControllerMappingProfile.Awake () (at Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerMappingProfile.cs:133)
```

The DefaultMixedRealityControllerMappingProfile.asset and the InputActions.MixedRealityControllerMappingProfile.asset need to be updated.  It can be updated by going here:

image

File: Documentation/CrossPlatform/OculusQuestMRTK.md [hunk 2]
```diff
@@ -35,6 +35,9 @@ work is 19.1. Older versions can be found from this [archive](https://developer.
 1. Navigate to Mixed Reality Toolkit > Utilities > Oculus > Configure CSC File for Oculus. Updating the csc file filters out the obsolete warnings produced by the Oculus Integration Assets. 
 The MRTK repo contains a csc file that converts warnings to errors, this conversion halts the MRTK-quest configuration process.
 
+1. Navigate to Mixed Reality Toolkit > Utilities > Oculus > Integrate Oculus Integration Unity Modules. Doing this will update the asmdefs with definitions and references needed for the
+relevant Oculus Quest code to function.
+
```
Comment (CDiaz-MS) [inline Documentation/CrossPlatform/OculusQuestMRTK.md:40]: nit: It helps to put a screenshot of the menu item to indicate the importance of the new step.  A picture of a step makes it hard to miss.

**Pull request merged**